src/server/plugins/engine/plugin.ts

```diff
@@ -100,155 +117,210 @@ export const plugin = {
-                  )
-              )
-            )
-          )
-          .code(200)
-      },
-      options: {
-        description: `${enabledString} Gets all published forms. Requires previewMode to be set to true. See runner/README.md for details on environment variables`
-      }
-    })
+    //     const { values } = forms[id]
+    //     return h.response(JSON.stringify({ id, values })).code(200)
+    //   },
+    //   options: {
+    //     description: `${enabledString} Gets a published form, by form id. Requires previewMode to be set to true. See runner/README.md for details on environment variables`
+    //   }
+    // })
+
+    // server.route({
+    //   method: 'get',
+    //   path: '/published',
+    //   handler: (request: Request, h: ResponseToolkit) => {
+    //     if (!previewMode) {
+    //       request.logger.error(
+    //         [`GET /published`, 'previewModeError'],
+    //         disabledRouteDetailString
+    //       )
+    //       throw Boom.unauthorized('publishing is disabled.')
+    //     }
+    //     return h
+    //       .response(
+    //         JSON.stringify(
+    //           Object.keys(forms).map(
+    //             (key) =>
+    //               new FormConfiguration(
+    //                 key,
+    //                 forms[key].name,
+    //                 undefined,
+    //                 forms[key].def.feedback?.feedbackForm
+    //               )
+    //           )
+    //         )
+    //       )
+    //       .code(200)
+    //   },
+    //   options: {
+    //     description: `${enabledString} Gets all published forms. Requires previewMode to be set to true. See runner/README.md for details on environment variables`
+    //   }
+    // })
 
     const queryParamPreHandler = async (
       request: Request,
       h: ResponseToolkit
     ) => {
       const { query } = request
-      const { id } = request.params
-      const model = forms[id]
-      if (!model) {
-        throw Boom.notFound('No form found for id')
-      }
-
+      const model = request.app.model
       const prePopFields = model.fieldsForPrePopulation
-      if (!Object.keys(query).length || !Object.keys(prePopFields).length) {
+      const queryKeysLength = Object.keys(query).length
+      const prePopFieldsKeysLength = Object.keys(prePopFields).length
+
+      if (queryKeysLength === 0 || prePopFieldsKeysLength === 0) {
         return h.continue
       }
+
       const { cacheService } = request.services([])
       const state = await cacheService.getState(request)
       const newValues = getValidStateFromQueryParameters(
         prePopFields,
         query,
         state
       )
+
       await cacheService.mergeState(request, newValues)
+
       return h.continue
     }
 
+    const loadFormPreHandler = server.app.model
+      ? async (request: Request, h: ResponseToolkit) =>
+          (request.app.model = server.app.model)
+      : async (request: Request, h: ResponseToolkit) => {
```

Comment:
just a style preference, I find this syntax a bit difficult to read. 

Could this just be one function with a quick return at the top?

```javascript
const loadFormPreHandler = async (request: Request, h: ResponseToolkit) => {
    if(server.app.model) { 
       // do this bit
    }
   
    // else
}
```

Reply:
It's only temporary and not what I'd usually do. At the time I thought we we're removing the capability to run a json file, but I wanted to retain the functionality while there are a bunch of tests that rely on it. If you don't like the ternary I can refactor now but it will all be getting re-done in the [clean-up ticket](https://dev.azure.com/defragovuk/DEFRA-CDP/_workitems/edit/376156).

Reply:
Resolved with 8d16917cb0bb396c2f281f89969c141c997cb1a7